lib/convert-to-typescript-component/index.tsx

```diff
@@ -32,33 +32,56 @@ export const convertBetterEasyToTsx = async ({
   const pn = normalizeManufacturerPartNumber(rawPn)
   const [cad_component] = su(circuitJson).cad_component.list()
 
-  // Derive pinLabels from easyeda json
-  const pinLabels: Record<string, string> = {}
-  for (const shape of betterEasy.dataStr.shape) {
-    if (shape.type === "PIN") {
-      const isPinLabelNumeric = /^\d+$/.test(shape.label)
-      const label = isPinLabelNumeric ? `pin${shape.label}` : shape.label
+  // Derive schPinArrangement from easyeda json
+  const pins = betterEasy.dataStr.shape.filter((shape) => shape.type === "PIN")
 
-      pinLabels[shape.pinNumber.toString()] = label
-    }
+  const hasStringPinNumbers = pins.some(
+    (pin) => typeof pin.pinNumber === "string",
+  )
+  let modifiedPins = pins
+  if (hasStringPinNumbers) {
+    modifiedPins = pins.map((pin, idx) => {
+      const originalPinNumber = pin.pinNumber.toString()
+
+      const newPinNumber = idx + 1
+
+      // If label is different from original pin number, create array with both pin number and label
+      // const newLabel =
+      //   originalPinNumber === pin.label
+      //     ? [pin.label]
+      //     : [originalPinNumber, pin.label]
+
+      return {
+        ...pin,
+        pinNumber: newPinNumber,
+        label: pin.label,
+      }
```

Comment:
Currently it looks like this, is it correct? 
![image](https://github.com/user-attachments/assets/ec87ae00-0c74-49a6-afc4-74ce833cc112)

Or should it look like this?
![image](https://github.com/user-attachments/assets/d6b8fb8d-580c-4b83-b151-351e405a7947)

Reply:
There's also duplicate `SH` label? Should we check for duplicate labels and add some unique characters?

Reply:
Nope duplicates are ok!!!

Reply:
@seveibar my bad, should've marked the PR draft. There's some ghost code as I wanted to confirm the structure. 
So, the first screenshot looks right?